jupyter_server/services/contents/handlers.py

```diff
@@ -208,28 +208,23 @@ async def post(self, path=""):
             raise web.HTTPError(400, "Cannot POST to files, use PUT instead.")
 
         model = self.get_json_body()
-        copy_from = model.get("copy_from")
-        if (
-            copy_from
-            and (
-                await ensure_async(cm.is_hidden(path))
-                or await ensure_async(cm.is_hidden(copy_from))
-            )
-            and not cm.allow_hidden
-        ):
-            raise web.HTTPError(400, f"Cannot copy file or directory {path!r}")
-
-        if model is not None:
+        if model:
             copy_from = model.get("copy_from")
+            if copy_from:
+                if not cm.allow_hidden and (
+                    await ensure_async(cm.is_hidden(path))
+                    or await ensure_async(cm.is_hidden(copy_from))
+                ):
+                    raise web.HTTPError(400, f"Cannot copy file or directory {path!r}")
+                else:
+                    await self._copy(copy_from, path)
+
             ext = model.get("ext", "")
             type = model.get("type", "")
             if type not in {None, "", "directory", "file", "notebook"}:
                 # fall back to file if unknown type
                 type = "file"
-            if copy_from:
-                await self._copy(copy_from, path)
-            else:
-                await self._new_untitled(path, type=type, ext=ext)
+            await self._new_untitled(path, type=type, ext=ext)
```

Comment:
Shouldn't this path only trigger if `copy_from` is false?

Reply:
Shoot, yeah good catch - looks like a follow-up is in order!

Reply:
Thanks @vidartf - I apologize for not catching this as well.